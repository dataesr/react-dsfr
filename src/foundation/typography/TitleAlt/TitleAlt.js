import PropTypes from 'prop-types';
import classnames from 'classnames';
import { CLASS_NAME_TYPE } from '../../../utils/variables';

/**
 *
 * @visibleName TitleAlt
 */
const TitleAlt = ({
  as, size, children, className,
}) => {
  const HtmlTag = `${as}`;
  return (
    <HtmlTag className={classnames({
      className,
      [`rf-display-${size}`]: size,
    })}
    >
      {children}
    </HtmlTag>
  );
};

TitleAlt.propTypes = {
  children: PropTypes.string.isRequired,
  /**
   * One of: string, object
   */
  className: CLASS_NAME_TYPE,
  /**
  * html tag to render
  */
  as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']),
  size: PropTypes.oneOf(['xl', 'lg', 'md', 'sm', 'xs']).isRequired,
};

TitleAlt.defaultProps = {
  as: 'h1',
  className: '',
};

export default TitleAlt;
