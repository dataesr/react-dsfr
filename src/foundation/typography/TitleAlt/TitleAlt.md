### Alternative titles

```jsx
<TitleAlt as="h1" size="xl">Title xl</TitleAlt>
```

```jsx
<TitleAlt as="h1" size="lg">Title lg</TitleAlt>
```

```jsx
<TitleAlt as="h1" size="md">Title md</TitleAlt>
```

```jsx
<TitleAlt as="h1" size="sm">Title sm</TitleAlt>
```

```jsx
<TitleAlt as="h1" size="xs">Title xs</TitleAlt>
```
