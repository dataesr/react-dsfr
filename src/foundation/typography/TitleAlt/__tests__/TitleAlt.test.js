import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import TitleAlt from '../TitleAlt';

const initialProps = {
  as: 'h2',
  size: 'xl',
  children: 'Title Alt xl',
};

describe('<TitleAlt />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(<TitleAlt {...initialProps}>{initialProps.children}</TitleAlt>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
