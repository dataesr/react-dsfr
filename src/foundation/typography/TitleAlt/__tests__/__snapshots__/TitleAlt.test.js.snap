// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleAlt /> renders correctly 1`] = `
<h2
  className="rf-display-xl"
>
  Title Alt xl
</h2>
`;
